src/unix/stream.c

```diff
@@ -1625,6 +1625,21 @@ void uv__stream_close(uv_stream_t* handle) {
 }
 
 
+/* Have stream block and then synchronously flush queued writes.
+ * This function works without an event loop.
+ * Intended to be used just prior to exit().
+ * Returns 0 on success, non-zero on failure.
```

Comment:
It's only considered to be successfully flushed if the write queue size for the uv stream is zero - nothing left to write.